source/extensions/stat_sinks/dog_statsd/config.h

```diff
@@ -9,6 +9,9 @@ namespace Extensions {
 namespace StatSinks {
 namespace DogStatsd {
 
+// DogStatsD compatible statsd sink
+const std::string DogStatsd = "envoy.stat_sinks.dog_statsd";
```

Comment:
sorry, you can't unfortunately make a static-init string like this.

see https://github.com/envoyproxy/envoy/blob/main/STYLE.md

> The Google C++ style guide points out that non-PoD static and global variables are forbidden. This includes types such as std::string. We encourage the use of the advice in the C++ FAQ on the static initialization fiasco for how to best handle this.

In this case I'd put this into config.h:
```
constexpr char DogStatsd[];
```
and in config.cc I'd put:
```
constexpr char DogStatsd[] = "envoy.stat_sinks.dog_statsd";
```
does that make sense? This is because char-arrays can be statically initialized by the compiler, and are not subject to init-order issues.

Reply:
...or you could retain the pattern for well_known_names but just have separate ones for each extension.

Reply:
Oh, I prefer the first way.

Reply:
> Constexpr variable declaration must be a definitionclang(invalid_constexpr_var_decl)

So I use `extern constexpr char DogStatsd[] = "envoy.stat_sinks.dog_statsd";` in config.h

Reply:
Sorry -- the pattern I gave you is for `const`. For `constexpr` it's simpler; just put the declaration and value in the header file.

https://www.learncpp.com/cpp-tutorial/sharing-global-constants-across-multiple-files-using-inline-variables/